app/models/restaurant.rb

```diff
@@ -0,0 +1,68 @@
+# == Schema Information
+#
+# Table name: restaurants
+#
+#  id                    :bigint(8)        not null, primary key
+#  address               :text
+#  cuisine               :string
+#  maximum_delivery_time :integer
+#  name                  :string
+#  rating                :decimal(2, 1)    default(0.0)
+#  tenbis                :boolean          default(FALSE)
+#  weeat_rating          :decimal(2, 1)    default(0.0)
+#  zomato_rating         :decimal(2, 1)    default(0.0)
+#  created_at            :datetime         not null
+#  updated_at            :datetime         not null
+#  zomato_restaurant_id  :integer
+#
+# Indexes
+#
+#  index_restaurants_on_zomato_restaurant_id  (zomato_restaurant_id)
+#
+
+class Restaurant < ApplicationRecord
+  validates :name, :address, :cuisine, :maximum_delivery_time, presence: true
+  validates :name, uniqueness: { scope: :address,
+    message: "can't have 2 restaurants with the same name in the same address" }, on: :create
+  validates :rating, :zomato_rating, :weeat_rating, numericality: { only_integer: false, greater_than_or_equal_to: 0, less_than_or_equal_to: 5 }
+
+  has_many :reviews
+
+  def recalculate_ratings!
+    calculate_and_set_weeat_rating
+
+    self.rating = if has_both_weeat_rating_and_zomato_rating?
```

Comment:
I know TOmer will have to say something about it... he doesn't like ifs

Reply:
I have to explain this to you face2face so you'll understand why I did this here and why there are 3 columns here for ratings. (1 for the rating imported from zomato, 1 for rating create on the weeat platform and 1 that averages both zomato and weeeat - and that's the one that will be displayed)

Reply:
The reason it partialy explained in the specs for this method.

Reply:
I'll be happy to hear it
Other than that there are better solutions for this flows that a large switch case, which makes the code not Open close at all https://www.eventhelix.com/RealtimeMantra/Object_Oriented/open_closed_principle.htm#.WvQh9dPRDYU